### PR TITLE
APP-1904: Add datadog integration

### PR DESCRIPTION
**Pull request by hugokallstrom**

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -292,4 +302,7 @@ dependencies {
   kaptDebug(libs.showkase.processor)
   "kaptStaging"(libs.showkase.processor)
   "kaptPullrequest"(libs.showkase.processor)
+
+  implementation("com.datadoghq:dd-sdk-android:1.8.1")
```
Comment (hugokallstrom) [inline app/build.gradle.kts:306]: Need to use libs here, but I get: 
```
Caused by: org.gradle.internal.typeconversion.UnsupportedNotationException: Cannot convert the provided notation to an object of type Dependency: org.gradle.accessors.dm.LibrariesForLibs$DatadogDdLibraryAccessors@4e80efe7.
The following types/formats are supported:
  - Instances of Dependency.
  - String or CharSequence values, for example 'org.gradle:gradle-core:1.0'.
  - Maps, for example [group: 'org.gradle', name: 'gradle-core', version: '1.0'].
  - FileCollections, for example files('some.jar', 'someOther.jar').
  - Projects, for example project(':some:project:path').
  - ClassPathNotation, for example gradleApi().

Comprehensive documentation on dependency notations is available in DSL reference for DependencyHandler type.
	at org.gradle.internal.typeconversion.ErrorHandlingNotationParser.parseNotation(ErrorHandlingNotationParser.java:57)
	at org.gradle.api.internal.artifacts.DefaultDependencyFactory.createDependency(DefaultDependencyFactory.java:60)
	at org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.create(DefaultDependencyHandler.java:173)
	at org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.doAddRegularDependency(DefaultDependencyHandler.java:199)
	at org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.doAdd(DefaultDependencyHandler.java:194)
	at org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.add(DefaultDependencyHandler.java:125)
	at org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.add(DefaultDependencyHandler.java:119)
	at org.gradle.kotlin.dsl.support.delegates.DependencyHandlerDelegate.add(DependencyHandlerDelegate.kt:55)
	at org.gradle.kotlin.dsl.ImplementationConfigurationAccessorsKt.implementation(Unknown Source)
	at Build_gradle$3.invoke(build.gradle.kts:306)
	at Build_gradle$3.invoke(build.gradle.kts:2)
	at org.gradle.kotlin.dsl.ProjectExtensionsKt.dependencies(ProjectExtensions.kt:178)
	at Build_gradle.<init>(build.gradle.kts:144)
	... 187 more
```
when I add `implementation(libs.datadog.dd)`. Any ideas why?

Comment (hugokallstrom) [reply app/build.gradle.kts]: that was it, thanks

**Review by StylianosGakis — APPROVED** [review]
Is the idea that we use this on top of what we already have in firebase as well?

File: app/src/main/java/com/hedvig/app/HedvigApplication.kt [hunk 2]
```diff
@@ -79,6 +87,40 @@ open class HedvigApplication : Application() {
...
+      .setNetworkInfoEnabled(true)
+      .setLogcatLogsEnabled(true)
+      .setDatadogLogsEnabled(true)
+      .setBundleWithTraceEnabled(true)
+      .build()
+
+    Timber.plant(DatadogTree(logger))
+
+    val monitor = RumMonitor.Builder().build()
+    GlobalRum.registerIfAbsent(monitor)
```
[the remark this answers is not in the document]
Comment (hugokallstrom) [reply app/src/main/java/com/hedvig/app/HedvigApplication.kt:123]: Datadog uses it in order to create user sessions, so you can look at funnels etc.

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -155,6 +156,8 @@ coreLibraryDesugaring = { module = "com.android.tools:desugar_jdk_libs", version
 coroutines-android = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-android", version.ref = "coroutines" }
 coroutines-core = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-core", version.ref = "coroutines" }
 coroutines-test = { module = "org.jetbrains.kotlinx:kotlinx-coroutines-test", version.ref = "coroutines" }
+datadog-sdk = { module = "com.datadoghq:dd-sdk-android", version.ref = "datadog" }
+datadog-timber = { module = "com.datadoghq:dd-sdk-android-timber", version.ref = "datadog" }
```
[the remark this answers is not in the document]
Comment (hugokallstrom) [reply gradle/libs.versions.toml:160]: True, I'll just run the Rum tracking for now. We can add custom logging later if we need it

File: app/build.gradle.kts [hunk 4]
```diff
@@ -10,12 +13,16 @@ plugins {
   id("kotlin-kapt")
   alias(libs.plugins.license)
   alias(libs.plugins.serialization)
+  id("com.datadoghq.dd-sdk-android-gradle-plugin")
```
[the remark this answers is not in the document]
Comment (hugokallstrom) [reply app/build.gradle.kts]: Think I got it!

**Pull request merged**